🐛 [BUG]: dev works, build fails -- it might be upstream -- discovery + workaround

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What package does the issue affects?

@nxtensions/astro

### Describe the bug

Hello + huge thank-you to leosvelperez for putting `@nxtensions/astro` together and especially for figuring out the fiddly bits with Nx like the ESM compatibility and packaging up that knowledge for the community!

## Error summary

I added the latest version 16.0.0 and ran it through a dev + then build to make sure that works before going too far... and... the build will fail with error.

The easiest to replicate consistently happens at the "generating static routes" step: 

`renderers.mjs` will not be found as "imported from `index.astro.mjs`". In the example output below take note of the paths and how the supposed import is going back too many directories (and actually looking far outside the project directory for `renderers.mjs`!).

Note that no page in the source actually imports directly from `renderers`: this is related to the "under the hood" of Astro.

### Issue confirmed even with a "virgin" Astro app created by the generator

The example output includes a few lines regarding tailwind not finding a definition for `content` as this seems related to the issue. 

In fact, tailwind + postcss has a valid config that actually mirrors what I have in an older project that is working that was created with this same nx astro plugin.

In investigating this issue, I ran an experiment with a totally virgin `@nxtensions/astro` generated app with no added configurations to confirm the problem and rule out the involvement of other configuration + plugins such as tailwind.

> FWIW in case it helps anyone: specifying the path to `configFile` in the `tailwind()` plugin setup in `astro.config.mjs` helps the tailwind config get "found" (no more content missing error), however this leads to new error output, presumably also related to paths not lining up. Specifying the configPath (if its in the standard location alongside astro config) isn't necessary with the fix/workaround noted below.

## Direct replication using Astro CLI (+ fix/workaround)

I looked into the code for the build executor and confirmed it uses the Astro CLI. 

I was able to replicate the issue using the Astro CLI directly.

See the comments above the commands below:

```sh
# this will seemingly build but take careful note of the output vs. the error output below
pnpm astro build --config apps/op/astro-ui/astro.config.mjs

# this exactly replicates the error seen when using `pnpm nx build ...` !!!
pnpm astro build --root apps/op/astro-ui --config astro.config.mjs

# after ensuring outDir, srcDir, and publicDir are all explicitly specified the error goes away with the following
pnpm astro build --config apps/op/astro-ui/astro.config.mjs
```

The [docs for Astro CLI](https://docs.astro.build/en/reference/cli-reference/) helped a bit but they aren't the most clear on how `--root` actually behaves. The docs entries for `outDir`, `srcDir`, and `publicDir` helped.

It seems like `--root` totally borks the import paths + module resolution, and specifying only a config that then explicitly defines all required paths leads to a solution workaround.

I haven't used the Astro CLI directly before so if this is a new or unexpected behaviour then this points to the issue being upstream. I hope this helps anyone that encounters this issue and helps lead to an extension/fix for this extension :) 

## Example error output

````txt
pnpm nx build op-astro-ui

> nx run op-astro-ui:build

12:18:09 PM [content] No content directory found. Skipping type generation.
12:18:09 PM [build] output target: static
12:18:09 PM [build] Collecting build info...
12:18:09 PM [build] Completed in 3.70s.
12:18:09 PM [build] Building static entrypoints...

warn - The `content` option in your Tailwind CSS configuration is missing or empty.
warn - Configure your content sources or your generated CSS will be missing styles.
warn - https://tailwindcss.com/docs/content-configuration
12:18:11 PM [build] Completed in 2.26s.

 building client
Completed in 1.08s.


 generating static routes
 error   Cannot find module '/home/username/projects/myproject/renderers.mjs' imported from /home/username/projects/m/yprojectdist/apps/op/src/pages/index.astro.mjs
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/username/projects/myproject/renderers.mjs' imported from /home/username/projects/myproject/dist/apps/op/src/pages/index.astro.mjs
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:324:11)
    at moduleResolve (node:internal/modules/esm/resolve:943:10)
    at defaultResolve (node:internal/modules/esm/resolve:1129:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36)
1

 >  NX   Ran target build for project op-astro-ui (12s)

    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]

```


### Expected behavior

Running `nx build <my-astro-app>` should work smoothly without error as it has done with previous versions of Astro and this plugin.

If `tailwind.config.cjs` is in the same directory as the Astro app in the nx monorepo under the apps folder and sitting alongside the `astro.config.mjs` file then there shouldn't be a need to specify the `configPath` argument to the `tailwind()` plugin.

### Steps to reproduce

See above.

### Link to minimal reproducible example

Easy to replicate on any nx with this extension.

### Environment

 >  NX   Report complete - copy this into the issue template

   Node   : 18.17.0
   OS     : linux-x64
   pnpm   : 8.6.11

   nx                 : 16.6.0
   @nx/js             : 16.6.0
   @nx/jest           : 16.6.0
   @nx/linter         : 16.6.0
   @nx/workspace      : 16.6.0
   @nx/cypress        : 16.6.0
   @nx/devkit         : 16.6.0
   @nx/esbuild        : 16.6.0
   @nx/eslint-plugin  : 16.6.0
   @nx/express        : 16.6.0
   @nx/next           : 16.6.0
   @nx/node           : 16.6.0
   @nx/plugin         : 16.6.0
   @nx/react          : 16.6.0
   @nx/storybook      : 16.6.0
   @nrwl/tao          : 16.6.0
   @nx/vite           : 16.6.0
   @nx/web            : 16.6.0
   nx-cloud           : 16.2.0
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @nxtensions/astro : 16.0.0
   ---------------------------------------

--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     "@types/tcp-port-used": "~1.0.1",
     "@typescript-eslint/eslint-plugin": "5.62.0",
     "@typescript-eslint/parser": "5.62.0",
-    "astro": "~2.9.6",
+    "astro": "~2.10.7",
     "commitizen": "~4.3.0",
     "eslint": "~8.46.0",
     "eslint-config-prettier": "~8.9.0",

--- a/packages/astro/src/generators/init/versions.ts
+++ b/packages/astro/src/generators/init/versions.ts
@@ -1,3 +1,1 @@
-// TODO(leo): temporary pin version to 2.9.6 (last working version) until
-// https://github.com/withastro/astro/issues/7934 is addressed.
-export const astroVersion = '2.9.6';
+export const astroVersion = '^2.10.7';

